### PR TITLE
Refactor processed mini blocks tracker

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Refactored processed mini blocks tracker

### CODE REVIEW

File: process/block/processedMb/processedMiniBlocks.go [hunk 1]
```diff
@@ -9,37 +9,37 @@ import (
...
 	mutProcessedMiniBlocks sync.RWMutex
 }
 
-// NewProcessedMiniBlocks will create a complex type of processedMb
-func NewProcessedMiniBlocks() *ProcessedMiniBlockTracker {
-	return &ProcessedMiniBlockTracker{
-		processedMiniBlocks: make(map[string]MiniBlocksInfo),
-	}
+// NewProcessedMiniBlocksTracker will create a processed mini blocks tracker object
+func NewProcessedMiniBlocksTracker() (*processedMiniBlocksTracker, error) {
```
Comment (iulianpascalau) [inline process/block/processedMb/processedMiniBlocks.go:28]: returning always nil for the error will result in a lot of dead code where we use this constructor. Please remove the error return param

File: process/sync/storageBootstrap/baseStorageBootstrapper.go [hunk 2]
```diff
@@ -164,11 +164,11 @@ func (st *storageBootstrapper) loadBlocks() error {
...
 
-	processedMiniBlocks := processedMb.NewProcessedMiniBlocks()
-	processedMiniBlocks.ConvertSliceToProcessedMiniBlocksMap(headerInfo.ProcessedMiniBlocks)
-	processedMiniBlocks.DisplayProcessedMiniBlocks()
+	processedMiniBlocksTracker, _ := processedMb.NewProcessedMiniBlocksTracker()
+	processedMiniBlocksTracker.ConvertSliceToProcessedMiniBlocksMap(headerInfo.ProcessedMiniBlocks)
+	processedMiniBlocksTracker.DisplayProcessedMiniBlocks()
 
-	st.blkExecutor.ApplyProcessedMiniBlocks(processedMiniBlocks)
+	st.blkExecutor.SetProcessedMiniBlocksTracker(processedMiniBlocksTracker)
```
Comment (iulianpascalau) [inline process/sync/storageBootstrap/baseStorageBootstrapper.go:171]: this is the only place were this set is used. We might convert the Set function into a Get function (only in shard/meta block processor) and call the `ConvertSliceToProcessedMiniBlocksMap` and `DisplayProcessedMiniBlocks` on what the Get function returns. We can easily drop all set functions on the tx coordinator and inner processors.

File: process/block/processedMb/processedMiniBlocks.go (hunk 1 above)
Comment (SebastianMarian) [reply process/block/processedMb/processedMiniBlocks.go:28]: Done

File: factory/coreComponents.go [hunk 3]
```diff
@@ -103,6 +103,7 @@ type coreComponents struct {
 	encodedAddressLen             uint32
 	arwenChangeLocker             common.Locker
 	processStatusHandler          common.ProcessStatusHandler
+	processedMiniBlocksTracker    process.ProcessedMiniBlocksTracker
```
Comment (iulianpascalau) [inline factory/coreComponents.go:106]: from linter:
`processedMiniBlocksTracker` is unused (structcheck)

File: factory/processComponents.go [hunk 4]
```diff
@@ -486,6 +488,8 @@ func (pcf *processComponentsFactory) Create() (*processComponents, error) {
 		return nil, err
 	}
 
+	processedMiniBlocksTracker := processedMb.NewProcessedMiniBlocksTracker()
```
Comment (iulianpascalau) [inline factory/processComponents.go:491]: :+1:

File: factory/coreComponents.go (hunk 3 above)
Comment (SebastianMarian) [reply factory/coreComponents.go]: Fixed

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 

@@                            Log scanner                                  @@
##                            refactor-processed-mini-blocks-tracker       ##
================================================================================
- Known Warnings              15
- New Warnings                3
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**